lib/src/main/java/io/ably/lib/rest/AblyRest.java

```diff
@@ -149,6 +152,51 @@ public void statsAsync(Param[] params, Callback<AsyncPaginatedResult<Stats>> cal
 	}
 
 	/**
+	 * Make a generic HTTP request for a collection of the given type
+	 * @param params query options: see Ably REST API documentation
+	 * for available options
+	 * @return a PaginatedResult of Stats records for the requested params
+	 * @throws AblyException
+	 */
+	public PaginatedResult<JsonElement> paginatedRequest(String method, String path, Param[] params, RequestBody body, Param[] headers) throws AblyException {
```

Comment:
We support `JSONRequestBody` and `ByteArrayRequestBody` in the library; I've added comments to that effect